### DISCUSSION TITLE
Simple warp example?

### DISCUSSION THREAD

**Discussion by napter** [elalish/manifold#409] [Q&A]
The examples that use warp are complex and difficult (for me) to understand. Can someone provide a simple example that uses warp that will run on manifoldcad.org?

**Comment by elalish** [comment]
You could try something like `yourCube.warp((v) => {v[0] += 1});` to shift your object by one in the x-direction. Of course translate is easier for that, but it gives the general idea. What do you want to use it for?

**Reply by Loosetooth** [reply]
Would you be open to have more examples added to manifoldcad.org ?
Perhaps it would be nice to have a dedicated manifold tutorial page. This could be for cpp and js/python/other bindings.

It took me some time, and some talking to other developers to get to understand the `warp` function for example. I just couldn't grasp it by looking at the source code and the existing manifoldcad examples. I think having some more basic code samples, or a tutorial, with a lot of comments would have helped a lot.

**Reply by pca006132** [reply]
Sure, we are open to PRs that can add more examples. The problem is that the operations we provide are very simple, so we are not sure what should we document. And for the *more advanced* APIs such as warp and LevelSet, the actual interface is also pretty simple (e.g. warp is just a vertex mapping, essentially), the problem is how you use the APIs.

**Reply by elalish** [reply]
Help with documentation would certainly be appreciated! And yes, I think a tutorial page might be more helpful at this point than just example files. I'm thinking we should add a new page to our wiki for this. Then again, I don't think Github allows arbitrary HTML in their Markdown, which will make it hard to have 3D interactive figures, which I think is what we'd really want for a good tutorial. For instance, I think it would be nice to output a GLB with two morph targets: before and after the warp, so you could animate the model to see the effect of the function. I'm working on improved GLB export now, so hopefully this will get easier. So perhaps we should just make another dedicated page under manifoldCAD.org.
